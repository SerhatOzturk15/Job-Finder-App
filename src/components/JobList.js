import React, {useEffect, useState} from "react";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import WorkIcon from "@material-ui/icons/Work";
import { deepOrange } from '@material-ui/core/colors';
import {
  Avatar,
  ListItemAvatar,
  ListItemText,
  ListItem,
  List
} from "@material-ui/core";
import {BASEURL} from './../util'

const useStyles = makeStyles(theme => ({
    root: {
      width: "100%",
      maxWidth: 500,
      margin: "0 auto",
      textDecoration: 'none',
      marginTop: 16
    },
    listItemText:{
        fontSize:'2em',
        textDecoration: 'none',
      },
      listItem:{
        "&:hover, &:focus": {
            backgroundColor: 'gray'
          }
      },
      avatar: {
        color: theme.palette.getContrastText(deepOrange[500]),
        backgroundColor: deepOrange[500],
      },
      link: {
        textDecoration: 'none',
         color: "black"
      },
  }));

const JobList = () => {

    const [jobs, setJobs] = useState([]);

    useEffect(() => {
        fetch(`${BASEURL}/getJobs`)
        .then(res => res.json())
        .then(result => {
            setJobs(result.data);
        })
        .catch(error => {
          console.log(error);
        });
    }, []);

  const classes = useStyles();
  return (
    <>
      <List className={classes.root} data-testid="job-list">
        {jobs.map((user, index) => (
          <h5 key={index}>
            <Link className={classes.link} to={`/job/${user.id}`}>
              <ListItem className={classes.listItem}>
                <ListItemAvatar>
                  <Avatar className={classes.avatar}>
                    <WorkIcon />
                  </Avatar>
                </ListItemAvatar>
                <ListItemText classes={{primary:classes.listItemText}} primary={user.title} />
              </ListItem>
            </Link>
          </h5>
        ))}
      </List>
    </>
  );
};

export default JobList;
