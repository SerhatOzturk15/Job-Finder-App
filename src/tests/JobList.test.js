import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import JobList from "../components/JobList";

const setup = (props = {}) => {
  const component = shallow(<JobList {...props} />);
  return component;
};

describe("JobList component", () => {
  describe("without props", () => {
    let component = setup();
    test("component should render", () => {
      const wrapper = component.find('[data-testid="job-list"]');
      expect(wrapper.length).toBe(1);
    });
    test("JobList snapshot should match", () => {
      const wrapper = component.find('[data-testid="job-list"]');
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("job-item should not render", () => {
      const wrapper = component.find('[data-testid="job-item"]');
      expect(wrapper.length).toBe(0);
    });
  });

  describe("with props", () => {
    const props = {
      jobs: [
        {
          id: 1,
          title: "Frontend Developer",
          description:
            "Comfortable with modern JS stack, experience with React.",
          employment_type: "full_time",
        },
        {
          id: 2,
          title: "Backend Developer",
          description: "Comfortable with Phyton, good knowledge of databases",
          employment_type: "full_time",
        },
      ],
    };
    let component = setup(props);
    test("component should render", () => {
      const wrapper = component.find('[data-testid="job-list"]');
      expect(wrapper.length).toBe(1);
    });
    test("2 job-item should render", () => {
      const wrapper = component.find('[data-testid="job-item"]');
      expect(wrapper.length).toBe(2);
    });
  });
});
