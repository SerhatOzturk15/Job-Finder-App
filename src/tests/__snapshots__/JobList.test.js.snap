// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList component without props JobList snapshot should match 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-1"
  data-testid="job-list"
/>
`;
